platform/distance.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "platform/measurement_utils.hpp"
+
+#include <string>
+
+namespace platform
+{
+class Distance
+{
+public:
+  enum class Units
+  {
+    Meters = 1,
+    Kilometers = 2,
+    Feet = 3,
+    Miles = 4,
+
+    // OSM-only
+    NauticalMiles = 5,
+    Inches = 6
+  };
+
+  Distance();
+
+  explicit Distance(double distanceInMeters);
+
+  explicit Distance(double distance, Units units);
+
+  [[nodiscard]] static Distance CreateFormatted(double distanceInMeters);
+
+  bool IsValid() const;
+
+  [[nodiscard]] Distance To(Units units) const;
+  [[nodiscard]] Distance ToPlatformUnitsFormatted() const;
+
+  double GetDistance() const;
+  Units GetUnits() const;
```

Comment:
BTW, following this _logic_, why this functions are not [[nodiscard]] ?

Reply:
Because these functions just return values and don't perform any time-consuming operations.